### PR TITLE
Fix change state

### PR DESCRIPTION
**Pull request by fmrico**
Hi @jhdcs 

I think I have found the bug that made fail `change_state()` and `get_current_state()` with a SIGSEGV. `State` was created via `from_raw()` method, so `rcl_lifecycle_state_init()` was not called (`State::new()` was never called). When we drop any `State`, `rcl_lifecycle_state_fini()` is always called getting a SIGSEGV

My fix is adding a flag to `State` to know whether it is created with `State::new()` or `from_raw()`. If it is created from `State::new()`, then `rcl_lifecycle_state_fini()` is called.

I have also added a couple of tests, and changed the return value of `change_state()`. It should be `State` , not if the transition has succeeded.

I hope you like the changes so we can continue with this branch :wink:

### CODE REVIEW

File: rclrs/src/lifecycle_node/state.rs [hunk 1]
```diff
@@ -107,11 +110,13 @@ impl State {
...
-        unsafe {
-            rcl_lifecycle_state_fini(state_handle.as_mut().unwrap(), &self.allocator).ok()?;
-        }
+        
+        //if !self.from_raw {
+            // SAFETY: By this point, we should have confirmed that state_handle still exists
+            unsafe {
+                rcl_lifecycle_state_fini(state_handle.as_mut().unwrap(), &self.allocator).ok()?;
+            }
+        //}
```
Comment (jhdcs) [inline rclrs/src/lifecycle_node/state.rs]:
```suggestion
        // SAFETY: By this point, we should have confirmed that state_handle still exists
        unsafe {
            rcl_lifecycle_state_fini(state_handle.as_mut().unwrap(), &self.allocator).ok()?;
        }
```

**Review by jhdcs — APPROVED** [review]
I apologize it took so long for me to get back to you on this - the notification got buried in my emails. In any case, sir, you are a SAINT! This has been driving me NUTS for so long!

The only minor issue I saw was a bit of commented code that was left in - you can either delete it now, or I can delete it after merge. In any case, thank you so very much for this!!!

File: rclrs/src/lifecycle_node/state.rs (hunk 1 above)
Comment (fmrico) [reply rclrs/src/lifecycle_node/state.rs]: Actually, it is just the opposite. I should have disabled the check for any test. I have just pushed a commit with this fixed :)

**Review by jhdcs — APPROVED** [review]

**Comment by jhdcs** [comment]
Thank you again!!! Merging this in!

**Pull request merged**